pyproject.toml

```diff
@@ -16,49 +22,100 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Bio-Informatics",
     "Intended Audience :: Science/Research"
 ]
-requires-python = ">= 3.10"
-requires = [
+dependencies = [
     "numpy",
     "pandas",
     "matplotlib",
     "seaborn",
     "h5py",
-    "anndata",
+    "anndata >= 0.10.8",
     "scanpy",
     "scikit-learn",
     "umap-learn",
-    "numba",
+    "numba >= 0.57",
     "protobuf",
     "tqdm",
     "mudata",
 ]
+dynamic = ["version"]
 
-[tool.flit.metadata.requires-extra]
+[project.urls]
+Documentation = "https://muon.readthedocs.io/en/latest/"
+Source = "https://github.com/scverse/muon"
+Home-page = "https://muon.scverse.org/"
+
+[project.optional-dependencies]
+dev = [
+    "setuptools_scm",
+]
 docs = [
-    "sphinx >= 4.0",
-    "pydata-sphinx-theme==0.8.1",
-    "sphinx-book-theme==0.3.3",
+    "sphinx",
+    "sphinx-book-theme",
+    "pydata-sphinx-theme",
     "readthedocs-sphinx-search",
     "nbsphinx",
-    "sphinx_automodapi"
-]
-atac = [
-    "pybedtools",
-    "pysam"
+    "sphinx_automodapi",
+    "recommonmark"
 ]
 test = [
     "pytest",
-    "flake8",
     "mofapy2",
+    "zarr",
+]
+# TODO: replace by a single depencency - chame
+atac = [
+    "pybedtools",
+    "pysam",
 ]
 
-[tool.flit.metadata.urls]
-Documentation = "https://muon.readthedocs.io/en/latest/"
-
-[tool.flit.sdist]
-exclude = [".github", "docs/build"]
-
+[tool.pytest.ini_options]
+python_files = "test_*.py"
+testpaths = [
+    "./tests", # unit tests
+]
 
 [tool.black]
```

Comment:
Just saw this randomly - there's no need to keep black when ruff-format exists (and is even configured here I think)